app/main.py

```diff
@@ -1,3 +1,66 @@
+from typing import Dict, Any
+
+
+def apply_armour(knight: Dict[str, Any]) -> int:
+    """Calculate total armour protection for a knight."""
+    return sum(piece["protection"] for piece in knight["armour"])
+
+
+def apply_weapon(knight: Dict[str, Any]) -> int:
+    """Calculate total power with the weapon applied."""
+    return knight["power"] + knight["weapon"]["power"]
+
+
+def apply_potion(knight: Dict[str, Any]) -> None:
+    """Apply potion effects to the knight, if any."""
+    if knight["potion"]:
```

Comment:
The check `if knight['potion']:` assumes that the 'potion' key is always present in the knight dictionary. It would be safer to use `if 'potion' in knight and knight['potion']:` to avoid potential KeyError if the 'potion' key is missing.